bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerHandleAdv.java

```diff
@@ -165,6 +165,11 @@ public void asyncAddEntry(final long entryId, final byte[] data, final int offse
      */
     @Override
     protected void doAsyncAddEntry(final PendingAddOp op, final ByteBuf data, final AddCallback cb, final Object ctx) {
+        if (op.entryId < 0 ) {
```

Comment:
when will this happen? I think this change volatiles the semantic of LedgerHandleAdv#asyncAddEntry. if the entry id is negative, we should reject this add with illegal arguments.